archive/shanghai-2019/Dockerfile

```diff
@@ -21,8 +21,8 @@ WORKDIR /home/node-dev
 ENV HOME /home/node-dev
 
 # Clone the code and build
-git clone --depth=1 https://github.com/nodejs/node.git
+RUN git clone --depth=1 https://github.com/nodejs/node.git
 WORKDIR /home/node-dev/node
 RUN python2 ./configure --ninja
```

Comment:
Why hardcode legacy Python?  Node.js currently prefers `python3` over `python2`.

Reply:
I've just failed with python3 yesterday... doesn't look into the error.

Reply:
Because the master does not currently work with python3 and we need to have a running Dockerfile by today.

Reply:
This Dockerfile failed??

Reply:
```
⋊> ~/D/g/node on master ⨯ python3 ./configure.py -C                                                                                                                                                         16:44:45
INFO: Using floating patch "tools/icu/patches/64/source/common/putil.cpp" from "tools/icu"
INFO: Using floating patch "tools/icu/patches/64/source/i18n/dtptngen.cpp" from "tools/icu"
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.
No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.
No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.
gyp: No Xcode or CLT version detected!
Error running GYP
⋊> ~/D/g/node on master ◦ python ./configure.py -C                                                                                                                                                          16:45:17
INFO: Using floating patch "tools/icu/patches/64/source/common/putil.cpp" from "tools/icu"
INFO: Using floating patch "tools/icu/patches/64/source/i18n/dtptngen.cpp" from "tools/icu"
INFO: configure completed successfully
```

I'll try to open an issue for it.

Reply:
BTW this is what currently happens if this command is switched to python3

```
Traceback (most recent call last):
  File "./configure", line 20, in <module>
    from distutils.spawn import find_executable
ModuleNotFoundError: No module named 'distutils.spawn'
```

Reply:
> 
> 
> BTW this is what currently happens if this command is switched to python3
> 
> ```
> Traceback (most recent call last):
>   File "./configure", line 20, in <module>
>     from distutils.spawn import find_executable
> ModuleNotFoundError: No module named 'distutils.spawn'
> ```

See https://github.com/nodejs/node/issues/30189.

Reply:
It still fails with 

```
Traceback (most recent call last):
  File "../../deps/v8/tools/gen-postmortem-metadata.py", line 693, in <module>
    load_objects();
  File "../../deps/v8/tools/gen-postmortem-metadata.py", line 375, in load_objects
    load_objects_from_file(filename, checktypes)
  File "../../deps/v8/tools/gen-postmortem-metadata.py", line 397, in load_objects_from_file
    for line in objfile:
  File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 2624: ordinal not in range(128)
```

after `python3-distutils`.

We need to give the organizers this Dockerfile ASAP. Is there a way to get this fixed in 1 hour

Reply:
Please stick with `python2` for now.

Reply:
@legendecas Thank you very much!  This Dockerfile has allowed me to find and propose fixes for nodejs/node#30218 and nodejs/node#30220.